<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::WindowModality::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1743</width>
    <height>999</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Повортная статика</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonStyle::ToolButtonTextBesideIcon</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">background-color: rgb(254, 254, 250);</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>4</number>
    </property>
    <property name="leftMargin">
     <number>4</number>
    </property>
    <property name="topMargin">
     <number>4</number>
    </property>
    <property name="rightMargin">
     <number>4</number>
    </property>
    <property name="bottomMargin">
     <number>4</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>10</number>
      </property>
      <property name="leftMargin">
       <number>10</number>
      </property>
      <property name="topMargin">
       <number>10</number>
      </property>
      <property name="rightMargin">
       <number>10</number>
      </property>
      <property name="bottomMargin">
       <number>10</number>
      </property>
      <item>
       <widget class="QFrame" name="left_panel">
        <property name="minimumSize">
         <size>
          <width>220</width>
          <height>0</height>
         </size>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="spacing">
          <number>4</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="spacing">
            <number>10</number>
           </property>
           <property name="leftMargin">
            <number>10</number>
           </property>
           <property name="topMargin">
            <number>10</number>
           </property>
           <property name="rightMargin">
            <number>10</number>
           </property>
           <property name="bottomMargin">
            <number>10</number>
           </property>
           <item>
            <widget class="QPushButton" name="btnInit">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>42</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Инициализация испытания</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnStatic1">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>42</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Статика 1</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnStatic2">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>42</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Статика 2</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnHand">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>42</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Ручное управление</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnCalibration">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>42</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Калибровка датчиков</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnProtocol">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>42</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Протокол/Результаты</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnArchive">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>42</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Архив</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnService">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>42</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Сервис</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Orientation::Vertical</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Policy::Expanding</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="btnStart">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>42</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Запуск</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnPause">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>42</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Пауза/Продолжить</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnStop">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>42</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Стоп</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnEmergencyReset">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>42</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Сброс аварии</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="spacing">
         <number>4</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QStackedWidget" name="pager">
          <property name="frameShape">
           <enum>QFrame::Shape::StyledPanel</enum>
          </property>
          <property name="lineWidth">
           <number>1</number>
          </property>
          <property name="currentIndex">
           <number>6</number>
          </property>
          <widget class="QWidget" name="pageHand">
           <layout class="QVBoxLayout" name="verticalLayout_19">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8" stretch="20,5">
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_5" stretch="1,10">
                <item>
                 <widget class="DashboardPanel" name="pageHand_pnlTopDashboard">
                  <layout class="QHBoxLayout" name="horizontalLayout_10">
                   <property name="spacing">
                    <number>6</number>
                   </property>
                   <property name="leftMargin">
                    <number>9</number>
                   </property>
                   <property name="topMargin">
                    <number>9</number>
                   </property>
                   <property name="rightMargin">
                    <number>9</number>
                   </property>
                   <property name="bottomMargin">
                    <number>9</number>
                   </property>
                  </layout>
                 </widget>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_9" stretch="12,3">
                  <item>
                   <widget class="PlotPanel" name="pageHand_pnlGraph">
                    <property name="frameShape">
                     <enum>QFrame::Shape::NoFrame</enum>
                    </property>
                    <layout class="QVBoxLayout" name="verticalLayout_7">
                     <property name="spacing">
                      <number>4</number>
                     </property>
                     <property name="leftMargin">
                      <number>0</number>
                     </property>
                     <property name="topMargin">
                      <number>0</number>
                     </property>
                     <property name="rightMargin">
                      <number>0</number>
                     </property>
                     <property name="bottomMargin">
                      <number>0</number>
                     </property>
                    </layout>
                   </widget>
                  </item>
                  <item>
                   <widget class="HandControlPanel" name="pageHand_control">
                    <property name="frameShape">
                     <enum>QFrame::Shape::StyledPanel</enum>
                    </property>
                    <property name="frameShadow">
                     <enum>QFrame::Shadow::Raised</enum>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
              <item>
               <widget class="LedDashboardPanel" name="pageHand_pnlRight"/>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="pageInit">
           <layout class="QVBoxLayout" name="verticalLayout_13">
            <property name="spacing">
             <number>0</number>
            </property>
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label">
              <property name="font">
               <font>
                <pointsize>24</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Инициализация испытания
В разработке...</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignmentFlag::AlignCenter</set>
              </property>
              <property name="wordWrap">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="pageStatic1">
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label_3">
              <property name="font">
               <font>
                <pointsize>24</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Статика 1
В разработке...</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignmentFlag::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="pageStatic2">
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label_4">
              <property name="font">
               <font>
                <pointsize>24</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Статика 2
В разработке...</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignmentFlag::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="pageCalibration">
           <layout class="QHBoxLayout" name="horizontalLayout_18">
            <item>
             <widget class="ServoCalibrationWidget" name="frCalibration">
              <property name="frameShape">
               <enum>QFrame::Shape::NoFrame</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Shadow::Plain</enum>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="pageProtocol">
           <layout class="QHBoxLayout" name="horizontalLayout_protocol">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label_protocol">
              <property name="font">
               <font>
                <pointsize>24</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Протокол/Результаты
В разработке...</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignmentFlag::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="pageArchive">
           <layout class="QHBoxLayout" name="horizontalLayout_archive">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label_archive">
              <property name="font">
               <font>
                <pointsize>24</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Архив
В разработке...</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignmentFlag::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="pageService">
           <widget class="QGroupBox" name="groupBox">
            <property name="geometry">
             <rect>
              <x>30</x>
              <y>90</y>
              <width>201</width>
              <height>181</height>
             </rect>
            </property>
            <property name="title">
             <string>Коэффициенты ПИД-регулятора</string>
            </property>
            <widget class="QWidget" name="layoutWidget">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>30</y>
               <width>171</width>
               <height>91</height>
              </rect>
             </property>
             <layout class="QFormLayout" name="formLayout_4">
              <item row="0" column="0">
               <widget class="QDoubleSpinBox" name="dsbKP">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>90</width>
                  <height>22</height>
                 </size>
                </property>
                <property name="font">
                 <font>
                  <pointsize>10</pointsize>
                 </font>
                </property>
                <property name="decimals">
                 <number>4</number>
                </property>
                <property name="singleStep">
                 <double>0.000100000000000</double>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>KP</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QDoubleSpinBox" name="dsbKI">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>90</width>
                  <height>22</height>
                 </size>
                </property>
                <property name="font">
                 <font>
                  <pointsize>10</pointsize>
                 </font>
                </property>
                <property name="decimals">
                 <number>4</number>
                </property>
                <property name="singleStep">
                 <double>0.000100000000000</double>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLabel" name="label_6">
                <property name="text">
                 <string>KI</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QDoubleSpinBox" name="dsbKD">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>90</width>
                  <height>22</height>
                 </size>
                </property>
                <property name="font">
                 <font>
                  <pointsize>10</pointsize>
                 </font>
                </property>
                <property name="decimals">
                 <number>4</number>
                </property>
                <property name="singleStep">
                 <double>0.000100000000000</double>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>KD</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QPushButton" name="btnSavePID">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>130</y>
               <width>171</width>
               <height>42</height>
              </rect>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>42</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Сохранить</string>
             </property>
            </widget>
           </widget>
          </widget>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1743</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Настройки</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuExit">
    <property name="title">
     <string>Выход</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>О программе</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuExit"/>
   <addaction name="menu"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionAbout">
   <property name="text">
    <string>О программе</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>LedDashboardPanel</class>
   <extends>QFrame</extends>
   <header>src.ui.widgets.hand_right_panel</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>DashboardPanel</class>
   <extends>QFrame</extends>
   <header>src.ui.widgets.hand_top_panel</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>PlotPanel</class>
   <extends>QFrame</extends>
   <header>src.ui.widgets.hand_graph_panel</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ServoCalibrationWidget</class>
   <extends>QFrame</extends>
   <header>src.ui.widgets.calibration_widget</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>HandControlPanel</class>
   <extends>QFrame</extends>
   <header>src.ui.widgets.hand_control_panel</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
